packages/mui-material/src/utils/types.ts

```diff
@@ -0,0 +1,33 @@
+import { SxProps } from '@mui/system';
+import { SlotComponentProps } from '@mui/base';
+
+export type SlotCommonProps = {
+  component?: React.ElementType;
+  sx?: SxProps;
+};
+
+export type SlotProps<
+  TSlotComponent extends React.ElementType,
+  TOverrides,
+  TOwnerState,
+> = SlotComponentProps<TSlotComponent, SlotCommonProps & TOverrides, TOwnerState>;
```

Comment:
Changes this implementation compared to Joy to use Base UI's type as a base (no pun intended).